replica/table.cc

```diff
@@ -705,6 +705,7 @@ table::try_flush_memtable_to_sstable(lw_shared_ptr<memtable> old, sstable_write_
         // priority inversion.
         auto post_flush = [this, old = std::move(old), &newtabs, f = std::move(f)] () mutable -> future<stop_iteration> {
             try {
+                co_await _compaction_manager.maybe_wait_on_current_compaction(this);
```

Comment:
Shouldn't we do this before flushing rather than after?

Reply:
Fair point, though if I read the code correctly, the variable naming is off which leads to some confusion.

The actual flushing happens on 

`co_await std::move(f);`

which is called after maybe_wait_on_current_compaction. That being said, it's called within the post_flush lambda, which is specifically created to run post flush actions in default_scheduling_group. So as a prerequisite, we should first move both calls out of that lambda (so flush is actually called in the correct scheduling group) and add error handling around flush. Flushing in the wrong group would also explain why statically limiting the shares for memtable_scheduling_group did nothing to slow it down in the test.